server/src/main/java/ovh/nixenos/tab/server/users/User.java

```diff
@@ -0,0 +1,256 @@
+   * default constructor to be used: taking username, first name, last name,
+   * role and password hash as arguments; active flag is by default set to True
+   * */
+  public void Customer(String username, String firstName, String lastName,
+                       String role, String password)
+      throws ConstructorCreationException {
+    try {
+      this.setUsername(username);
+      this.setFirstName(firstName);
+      this.setLastName(lastName);
+      this.setPassword(password);
+      this.setRole(role);
+    } catch (InvalidArgumentException eInvalidArgumentExceptionInstance) {
+      System.err.println(eInvalidArgumentExceptionInstance.getMessage());
+      throw new ConstructorCreationException();
+    }
+  }
+
+  /**
+   * setter method for username; throws InvalidArgumentException on null or
+   * empty argument
+   * */
+  private void setUsername(String username) throws InvalidArgumentException {
+    if (username == null) {
+      throw new InvalidArgumentException("Username cannot be null");
+    }
+    if (username.isBlank()) {
+      throw new InvalidArgumentException("Username cannot be empty");
+    }
+    this.username = username;
+  }
+
+  /**
+   * setter method for user's first name; throws InvalidArgumentException on
+   * null or empty argument
+   * */
+  private void setFirstName(String firstName) throws InvalidArgumentException {
+    if (firstName == null) {
+      throw new InvalidArgumentException("User's name cannot be null");
+    }
+    if (firstName.isBlank()) {
+      throw new InvalidArgumentException("User's name cannot be empty");
+    }
+    this.firstName = firstName;
+  }
+
+  /**
+   * setter method for user's last name; throws InvalidArgumentException on null
+   * or empty argument
+   * */
+  private void setLastName(String lastName) throws InvalidArgumentException {
+    if (lastName == null) {
+      throw new InvalidArgumentException("User's last anme cannot be null");
+    }
+    if (lastName.isBlank()) {
+      throw new InvalidArgumentException("User's last name cannot be empty");
+    }
+    this.lastName = lastName;
+  }
+
+  /**
+   * setter method for users password, throws InvalidArgumentException
+   * */
+  private void setPassword(String password) throws InvalidArgumentException {
+    if (password == null) {
+      throw new InvalidArgumentException("Password cannot be null");
+    }
+    if ((password.isBlank())) {
+      throw new InvalidArgumentException("Password cannot be blank");
+    }
+    this.passwordHash = hashPasswordString(password);
+  }
+
+  /**
+   * method for hashing user's password
+   * @TODO implement hashing
+   * */
+  private String hashPasswordString(String password) { return password; }
+
+  /**
+   * setter method for user's role; throws InvalidArgumentException
+   * */
+  private void setRole(String role) throws InvalidArgumentException {
+    ArrayList<String> possible_roles =
+        new ArrayList<String>(Arrays.asList("admin, manager, worker"));
```

Comment:
as I mentioned in Personel.java file above, maybe it's a good idea to squash this possible_roles into some enum.
some of the reasons:
- if we want to check some roles later it would require to write again this raw arraylist<string>
- if we want to add some role it is better to do it in separated file, not in method of a class